client/src/modules/Auth/SignIn/reducers.js

```diff
@@ -0,0 +1,42 @@
+import {
+  SIGNIN_REQUEST,
+  SIGNIN_SUCCESS,
+  SIGNIN_FAILURE
+} from './const';
+
+const user = (localStorage.getItem('user') !== 'undefined') ?
+  JSON.parse(localStorage.getItem('user')) :
+  undefined;
+
```

Comment:
You can use "or" logical operator instead of ternary operator.
E.g. const val = localStorage.getItem('value')  || {} (or what You need).